postgres-appliance/Dockerfile.build

```diff
@@ -0,0 +1,421 @@
+    && rm -rf /var/lib/apt/lists/* /var/cache/debconf/* /usr/share/doc /usr/share/man /usr/share/locale \
+    && find /var/log -type f -exec truncate --size 0 {} \;
+
+ADD dependencies/debs /builddeps
+
+ARG WITH_PERL=false
+ARG PGOLDVERSIONS="9.3 9.4 9.5 9.6"
+
+# Install PostgreSQL, extensions and contribs
+ENV PGVERSION=10 \
+    PGOLDVERSIONS="$PGOLDVERSIONS" \
+    POSTGIS_VERSION=2.4 \
+    BG_MON_COMMIT=9412922efba655491ccebbcbfc5b7308d79b90c4 \
+    DECODEBUFS_COMMIT=d08a75593c68897600e10f5250133337624bcf63 \
+    HYPOPG=1.1.0 \
+    SET_USER=REL1_4_0 \
+    PLPGSQL_CHECK=v1.2.2 \
+    PLPROFILER=REL3_2 \
+    TIMESCALEDB=0.7.1 \
+    PGEXTWLIST=v1.6
+
+RUN export DEBIAN_FRONTEND=noninteractive \
+    && set -ex \
+    && apt-get update \
+    && cd /builddeps \
+    && BUILD_PACKAGES="devscripts build-essential equivs gcc automake autogen libtool cmake pgxnclient libedit-dev zlib1g-dev libssl-dev libselinux1-dev libkrb5-dev libxslt1-dev libxml2-dev libpam0g-dev libcurl4-openssl-dev pkg-config libevent-dev git" \
+    && if [ "$DEMO" = "true" ]; then \
+        PGOLDVERSIONS=""; \
+    else \
+        BUILD_PACKAGES="$BUILD_PACKAGES libprotobuf-c-dev liblwgeom-dev libproj-dev" \
+        && if [ "$WITH_PERL" != "true" ]; then \
+            apt-get install -y equivs \
+            && version=$(apt-cache show perl | sed -n 's/^Version: //p' | sort -rV | head -n 1) \
+            && echo "Section: misc\nPriority: optional\nStandards-Version: 3.9.2\nPackage: perl\nReplaces: perl-base\nVersion: $version\nDescription: perl" > perl \
+            && equivs-build perl \
+            && dpkg -i perl*.deb; \
+        fi \
+
+        && dpkg -i libcgal[^-]*_amd64.deb \
+                    libgdal[^-]*_amd64.deb \
+                    libsfcgal[^-]*_amd64.deb \
+                || apt-get -y -f install \
+        && apt-get install -y libprotobuf-c1 \
+
+        && curl -s -L https://github.com/debezium/postgres-decoderbufs/archive/$DECODEBUFS_COMMIT.tar.gz | tar xz; \
+    fi \
+
+    && apt-get install -y libcurl3 libevent-2.0 libevent-pthreads-2.0 python3.5 $BUILD_PACKAGES \
+
+    # prepare 3rd sources
+    && curl -s -L https://github.com/CyberDem0n/bg_mon/archive/$BG_MON_COMMIT.tar.gz | tar xz \
+
+    && git clone -b $HYPOPG https://github.com/dalibo/hypopg.git \
+
+    && git clone -b $SET_USER https://github.com/pgaudit/set_user.git \
+
+    && git clone -b $PLPGSQL_CHECK https://github.com/okbob/plpgsql_check.git \
+
+    && git clone -b $PLPROFILER https://bitbucket.org/openscg/plprofiler.git \
+
+    && git clone -b $TIMESCALEDB https://github.com/timescale/timescaledb.git \
+
+    && git clone -b $PGEXTWLIST https://github.com/dimitri/pgextwlist.git \
+    && perl -ne 'print unless /PG_TRY/ .. /PG_CATCH/' pgextwlist/pgextwlist.c > pgextwlist.c.f \
+    && egrep -v '(PG_END_TRY|EmitWarningsOnPlaceholders)' pgextwlist.c.f > pgextwlist/pgextwlist.c \
+
+    && git clone https://github.com/pgq/skytools-legacy.git \
```

Comment:
you can do it with git clone --recurse-submodules

Reply:
Fixed. Thank you for the tip.